01.multithreading.v.1.2/MultiThreading.Task4.Threads.Join/Program.cs

```diff
@@ -26,9 +27,47 @@ static void Main(string[] args)
 
             Console.WriteLine();
 
-            // feel free to add your code
+            // Part A
+            Console.WriteLine("Starting Part A: Thread class with Join");
+            CreateThreads(10, 10);
+
+            // Part B
+            Console.WriteLine("\nStarting Part B: ThreadPool class with Semaphore");
+            CreateThreadsWithThreadPool(10, 10);
 
             Console.ReadLine();
         }
+
+        // Part A: Using Thread class
+        static void CreateThreads(int maxThreads, int state)
+        {
+            if (maxThreads <= 0) return;
+
+            Thread thread = new Thread(() =>
+            {
+                Console.WriteLine($"Thread with state: {state}");
```

Comment:
It looks like you did the workaround rather passing the sate, right now I see that your thread is capturing the int value and copies its value to the thread space, however intension of the task is to us share object state in each thread and working with it. 
One of the implementation option is, however it is also not complete, because you need return result from thread that  subtract value,  review videos and microsoft documentation 

https://learn.microsoft.com/en-us/dotnet/standard/threading/creating-threads-and-passing-data-at-start-time

Reply:
I hope my adjustments meet your expectations